Fix issue with CHAR matching in the WHERE clause. (#16770)
crate/crate: docs/appendices/release-notes/5.8.5.rst

Fixes
=====

- Fixed an issue that would lead to wrong results being returned when trying to

>   match a column of type :ref:`CHAR<type-char>` with empty string literals, or
>   literals containing only whitespaces (``''``, ``'   '``, etc.) in the

Reviewer: Isn't it irrelevant what chars are used?
```suggestion
  compare a column of type :ref:`CHAR<type-char>` with a column any character varying type 
  in the
```

Author: yes, I will rephrase, I realized when writing the commit message, but forgot.
